database/buildSchema/43_views_functions_triggers.sql

```diff
@@ -1542,164 +1542,377 @@ $$
 LANGUAGE sql
 STABLE;
 
+-- Does the "heavy lifting" to deduce the assigners/reviewers list for a single
+-- application as well as the available actions for a particular
+-- reviewer/assigner for that application. Used to populate the
+-- `application_reviewer_action` table and the reviewer/assigner lists on each
+-- application whenever an event occurs that would change these values
+CREATE OR REPLACE FUNCTION single_application_detail (applicationId int, userId int DEFAULT 0)
```

Comment:
This function is a pure copy/paste from the existing list function. Differences are
- only for a single application (passed in `applicationId` parameter, and additional `WHERE` clause).
- removed all columns other than these 4 complicated ones -- the others are all just joined in the list query as before.

Reply:
Still has a `DEFAULT` value since when generating the assigners and reviewers lists, we don't need to specify a user.